Fixed bug in ELC

KonradBreitsprecher opened the pull request:
Fixes ELC part of #1559 and probalby #229  

 - Found small but fatal bug in elc.cpp. Also I could no longer reproduce the error in ELCIC and it agrees with MMM2D with two particles and metallic plates.

KonradBreitsprecher committed:
added dielectric contrast in elc interface

KonradBreitsprecher committed:
logs in elc

KonradBreitsprecher committed:
fixed bug in elc scycache

KonradBreitsprecher committed:
Merge branch 'python' of https://github.com/espressomd/espresso into konrad/elcic_fix

KaiSzuttor reviewed (commented):
please remove debug printfs

fweik commented on src/python/espressomd/particle_data.pyx:
```diff
@@ -165,7 +165,9 @@ cdef class ParticleHandle(object):
 
             unfold_position(ppos, img)
-            return np.array([ppos[0], ppos[1], ppos[2]])
+            ret = np.array([ppos[0], ppos[1], ppos[2]])
```
This should not be in here.

fweik commented on src/core/elc.cpp:
```diff
@@ -1316,6 +1327,7 @@ int ELC_set_params(double maxPWerror, double gap_size, double far_cut,
     elc_params.far_calculated = 0;
   } else {
     elc_params.far_calculated = 1;
+    printf("ELC_set_params tune\n");
```
Left over printf

fweik commented on src/core/elc.cpp:
```diff
@@ -1230,14 +1233,22 @@ void ELC_init() {
+    p3m.params.additional_mesh[2] = 0;
   }
+  
+  printf("far_cut %f\n far_cut2 %f\n h %f\n space_layer %f\n space_box %f\n far_calculated %d\n addmesh %f %f %f\n",elc_params.far_cut,elc_params.far_cut2, elc_params.h, elc_params.space_layer, elc_params.space_box, elc_params.far_calculated, p3m.params.additional_mesh[0],p3m.params.additional_mesh[1],p3m.params.additional_mesh[2]);
```
Left over printf

fweik commented on src/core/elc.cpp:
```diff
@@ -1230,14 +1233,22 @@ void ELC_init() {
 
   if (elc_params.far_calculated && (coulomb.method == COULOMB_ELC_P3M &&
                                     elc_params.dielectric_contrast_on)) {
+    printf("ELC_init tune\n");
```
Left over printf

fweik commented on src/core/elc.cpp:
```diff
@@ -1196,6 +1198,7 @@ int ELC_sanity_checks() {
 }
 
 void ELC_init() {
+  printf("ELC_init\n");
```
Left over printf

fweik commented on src/core/elc.cpp:
```diff
@@ -1137,6 +1137,7 @@ double ELC_energy() {
 }
 
 int ELC_tune(double error) {
+  printf("ELC_tune\n");
```
Left over printf

fweik reviewed (changes requested):
Looks good to me, aside from the obvious stuff that needs to be changed.

KonradBreitsprecher committed:
cleanup

KonradBreitsprecher committed:
Update elc.cpp

Fixed wrong error estimate in ELC_tune() (compare to arnold02d eq. 16 and arnold02c eq. 18).
